### PR TITLE
Gun Additions: .22 Auto

### PR DESCRIPTION
**Pull request by DVDPlayerOfDiscordFame**
In this PR, I've been developing (attempting to) a new ammo-type and two new guns to the game. The ammo type in question is .22 auto-- a new pistol ammo that deals less damage than the default .35 auto (14 pierce vs. 16 pierce). The reason behind this is to make an ammo that is weak enough to enable the use of a faster-firing weapon without breaking balance, as well as develop an Absolute Dogshit Pistol. Included with the .22 auto is every ammunition sub-type as well with appropriate damage values.

The weapons added are as follows:
**the Mk 93** -- a Security Machine Pistol, fulfilling the role of a sort of "downgraded WT550" that allows security members to more often use the classic automatic weapon + shield loadout, as well as have an automatic weapon that is more easily attainable for Security than the WT550-- at the cost of having worse accuracy, a smaller magazine, a DPS squarely between the Mk 53 and the WT550, and less damage-per-shot. The current intent is to have this weapon be produced via Secfab, by being unlocked with research, but could also just be a round-start armory weapon.

**The Zipper** -- A dogshit maintenance pistol, boasting the worst accuracy and DPS out of every fire arm in the game by a WIDE margin. The only up-side to this piece of shit is that it's a tiny object and takes up one square (intended to). With its small magazine size, you would have to land EVERY shot in order to crit a full-health individual, which is a tall order because you would need to be point-blank the entire time and deal with a 3 shots/second rate of fire.

This PR currently only contains the assets for the weapons. Inclusions in research, the secfab, cargo orders, and/or maintenance drops have not been added yet.

**Changelog**
:cl:
- add: .22 Auto Ammunition (base/practice/incendiary/uranium)
- add: Mk 93 Machine Pistol
- add: Zipper Compact Pistol

### CODE REVIEW

File: Resources/Textures/_Impstation/Objects/Weapons/Guns/Pistols/zipper.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            "name": "icon"
+        },
+        {
+            "name": "base"
+        },
+        {
+            "name": "bolt-open"
+        },
+        {
+            "name": "mag-0"
```
Comment (hivehum) [inline Resources/Textures/_Impstation/Objects/Weapons/Guns/Pistols/zipper.rsi/meta.json:20]: this needs to be mag and not mag-0. either that or rename mag.png to mag-0

File: Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/Ammunition/Boxes/Lpistol.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+- type: entity
+  abstract: true
+  parent: BaseItem
```
Comment (hivehum) [inline Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/Ammunition/Boxes/Lpistol.yml:3]: same deal as the lpistol comment. could make this parent a different ammo container. etc etc etc. the parent system kind of fucks crazy once you learn it

**Review by hivehum — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/LightPistols/Lpistols.yml [hunk 3]
```diff
@@ -0,0 +1,143 @@
+- type: entity
+  name: BaseLPistol
+  parent: BaseItem
```
[the remark this answers is not in the document]
Comment (DVDPlayerOfDiscordFame) [reply Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/LightPistols/Lpistols.yml:3]: For the Zipper you mean? Or probably for both? My only concern with making any sort of direct link to the vanilla versions is the potential that a future change down the road from wizden fucks the ones I made as well.

[the remark this answers is not in the document]
Comment (hivehum) [reply Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/LightPistols/Lpistols.yml:3]: okay on second look, you mostly copied and pasted from BaseWeaponPistol, right? parent that and only add components you want to differ from it. you do run the risk of upstream changing BaseWeaponPistol but they're exceedingly unlikely to do something that is a drastic issue for your guns

[the remark this answers is not in the document]
Comment (DVDPlayerOfDiscordFame) [reply Resources/Prototypes/_Impstation/Entities/Objects/Weapons/Guns/LightPistols/Lpistols.yml:3]: I can do that I think. Changing BaseWeaponPistol is probably only something they would do if they did a massive weapons rework, and in that event I'd probably be fucked anyways.

**Pull request closed without merging**